openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClient.java

```diff
@@ -354,6 +354,10 @@ public Response search(SearchRequest request) throws IOException {
       searchSourceBuilder.sort(request.getSortFieldParam(), SortOrder.fromString(request.getSortOrder()));
     }
 
+    if (request.getIndex().equalsIgnoreCase("glossary_term_search_index")) {
```

Comment:
why not add this to GlossaryTermIndex.java instead of hard-coding like this?

Reply:
we need to add this query at the time when query api is called to fetch list .. thats why i thought and added it here..

Reply:
better to do it is to push down to Index classes no? pass the request and each class will add additonal things to request if required if not returns the same.
You can add default implementatoin in searchIndex to return the passed in request and override where its needed